### PR TITLE
Support for real SSL certificates at the PZH Web Server.

### PR DESCRIPTION
**Pull request by jplyle**
http://jira.webinos.org/browse/WP-396

Step one in WP-396 to remove our dependency on self-signed certificates.  
This change is necessary for pzh.webinos.org and other deployments.

Jira issue: WP-396

### CODE REVIEW

File: webinos/core/manager/certificate_manager/lib/certificate.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ var Certificate = function () {
     this.cert         = {};
     this.cert.internal= {};
     this.cert.external= {};
-    this.cert.internal= {master: {}, conn: {}, web: {}};
+    this.cert.internal= {master: {}, conn: {}, webclient: {}, webssl: {}};
```
Comment (habibvirji) [inline webinos/core/manager/certificate_manager/lib/certificate.js]: Would prefer if condition for webclient and webssl to add when it is PzhWS or PzhSSL

Comment (jplyle) [inline webinos/core/manager/certificate_manager/lib/certificate.js]: At this point in the code, there's no knowledge of whether it is a PZH, PZP or PZPWS invoking this function.

Where would I be putting the 'if' condition?  Sorry if this is a stupid question, I was just following the existing style of the code.

Comment (habibvirji) [inline webinos/core/manager/certificate_manager/lib/certificate.js]: Agreed.. you could just keep it master and conn here and in getSelfSignedCert, add if condition to add webclient and webssl in the internal structure.

**Comment by TobyEalden** [comment]
Out of interest - will this prevent people from running a PZH if they don't have a properly signed cert?

**Comment by jplyle** [comment]
Nope - it's just a few extra fields so that people can use a real certificate if they want to.  It'll all work as per normal otherwise.

**Comment by jplyle** [comment]
Hi @habibvirji - does this work?

**Pull request merged**